README.markdown

```diff
@@ -270,98 +251,52 @@ your Google Compute Engine project. So long as this chain remains trustworthy, t
-use that as the instance's boot disk. If no such disk exists, a new persistent
-disk will be created.
-
-When you delete an instance, the persistent disk will *not* be deleted. This
-provides the default behavior of persisting your data between instance
-termination and re-creation. If you truly want to delete a persistent disk,
-you must do so explicitly with it's own type-block and `ensure => absent` attribute.
-
 ### Enable/Disable Live Instance Migration
 
 Your Compute Engine instances, by default, will enable live migration.  In
 the event that Google needs to perform a datacenter maintenance, your instance
 will be automatically migrated to a new location without visible impact.
-This feature can be disabled by setting `on_host_migration` to `false`.
+This feature can be disabled by setting `maintenance_policy` to `TERMINATE`.
 
 ### Classifying resources
 
-`gce_instance` resources can also contain classification data to be used by the startup scripts described below.
+In addition to creating instances with `gce_instance`, you may pass additional parameters to configure and classify the instance. The work is done during instance creation by a bootstrap script. The module includes a scripts to configure both open source Puppet and Puppet Enterprise.
+
+In the `gce_instance` resource, you may provide the following parameter to choose a startup script. You can use any executable script that's located in the gce_compute modules files directory and can be interpreted by the OS GCE provisisions.
+
+```puppet
+   startupscript => 'puppet-community.sh'
+   startupscript => 'puppet-enterprise.sh'
+   startupscript => 'script_to_use.sh'
+```
 
 The classification is currently only supported by running `puppet apply`
-during the bootstrapping process of the created instances and can be done
-with the `enc_classes` parameter that utilizes
-[External Node Classifiers](http://docs.puppetlabs.com/guides/external_nodes.html)
-or by passing in the contents of a manifest file with the `manifest` parameter.
-If *both* parameters are specified, both will be applied to the instance with
-ENC first followed by the manifest file.
+during the bootstrapping process of the created instances
+by passing in the contents of a manifest file with the `manifest` parameter.
 
 Classification is specified with the following `gce_instance` parameters:
 
-* `modules` - List of modules that should be installed from the
+* `puppet_master` - Hostname of the puppet master instance that the
+  agent instance must be able to resolve. If this parameter is specified, then it is used as the `server` parameter in
+`puppet.conf`.
+* `puppet_service` - `absent` or `present`; if this parameter is specified, then the puppet service is automatically started
+on the managed instance and set to restart on boot (in `/etc/default/puppet`).
+* `puppet_manifest` - A string to pass in as a local manifest file and applied during
```

Comment:
This suggests that this is both a puppet manifest (in the first part) and an executable script in the second. I'm unclear which it is.